### DISCUSSION TITLE
URL API to open files

### DISCUSSION THREAD

**Discussion by gorshkov-leonid** [eclipse-theia/theia#8292] [General] [answered]
Is there any way to specify dynamically via URL which file should be opened?

**Comment by vince-fugnitto** [comment, marked as answer]
@gorshkov-leonid if you are referring to opening a file by URL through the browser address bar (https://github.com/eclipse-theia/theia/issues/4604) then it is not yet supported. At the moment, only workspaces can be opened through the browser address bar. 

If this is not your use-case can you please clarify further?

**Reply by gorshkov-leonid** [reply]
@vince-fugnitto Thank you, I am looking forward to this solution. But now I need some workaround. 
As suggested here https://github.com/eclipse/che/issues/17551, I can open a file using API theia/vscode or predefined command, but I do not know how to get URL to implement plugin.

The use case is links sharing.
